lib/shopify_api/resources/assigned_fulfillment_order.rb

```diff
@@ -1,6 +1,8 @@
 module ShopifyAPI
   class AssignedFulfillmentOrder < Base
     def self.find(scope, *args)
+      FulfillmentOrder.version_validation!
+
       assigned_fulfillment_orders = super(scope, *args)
       assigned_fulfillment_orders.map { |afo| FulfillmentOrder.new(afo.attributes) }
     end
```

Comment:
Version `AssignedFulfillmentOrder.initialize`--better to report errors sooner than later.

Reply:
I tried it, `find` and `all` do not actually go to `initialize`. It looks like the active resource uses a parser to initialize the object, it does not use the resource see:
https://github.com/rails/activeresource/blob/5038de5a0f6079d368fb8a80cc648ed93fa777e4/lib/active_resource/base.rb#L1103

Where is `load` call the resource initialize:
https://github.com/rails/activeresource/blob/5038de5a0f6079d368fb8a80cc648ed93fa777e4/lib/active_resource/base.rb#L1457

So in our case, we don't actually `new` an object. So I decided to only control it here.

Reply:
A client library isn't just to protect the endpoints of an API, it's to make the developers workflow better. We should try to handle semi-reasonable usages even if they don't ultimately reach an endpoint. Creating an AssignedFulfillmentOrder with `.new` may seem reasonable if they're not sure what the correct path should be and there's no reason to delay reporting errors.